python/tank/platform/environment.py

```diff
@@ -878,3 +880,262 @@ def copy_apps(self, src_engine_name, dst_engine_name):
         self.__write_data(self._env_path, data)
         # sync internal data with disk
         self._refresh()
+
+
+    ############################################################################
+    # Methods specific to updating all environment settings to full or sparse
+
+    def make_sparse(self, debug=False):
+        """
+        Make the settings defined in this environment sparsely populated.
+
+        Any setting defined in the environment that matches the default value
+        specified will be removed.
+
+        :param debug: If True, include debug comments on lines using a
+            non-default value.
+        """
+
+        self.__update_all_settings(UpdateAllSettingsFormat.SPARSE, debug)
+
+    def make_full(self, debug=False):
+        """
+        Make the settings defined in this environment fully populated.
+
+        Any setting not specified in the environment will be populated with the
+        default value.
+
+        :param debug: If True, include debug comments on lines using a default
+            valued.
+        """
+
+        self.__update_all_settings(UpdateAllSettingsFormat.FULL, debug)
+
+    def _update_settings_sparse(self, schema, settings, engine_name=None,
+        manifest_file=None, debug=False):
+        """
+        Given a schema and settings, make sure that the settings are sparse.
+
+        Iterate over the schema key's default values and remove any settings
+        that use the default.
+
+        :param schema: A schema defining types and defaults for settings.
+        :param settings: A dict of settings to sparsify.
+        :param engine_name: The name of the engine currently running if there
```

Comment:
this doc strings is confusing for me. "currently running" implies to me that you are running the code from within that engine, where it's more of a scope right? I think of it as a hierarchy: env > engine > app - could it be called someting like parent engine or current engine or something like that?